text/0000-global-allocators.md

```diff
@@ -240,9 +252,47 @@ The allocator APIs could be simplified to a more "traditional"
 malloc/calloc/free API at the cost of an efficiency loss when using allocators
 with more powerful APIs.
 
+The global allocator could be an instance of the `Allocator` trait. Since that
+trait's methods take `&mut self`, things are a bit complicated however. The
+allocator would most likely need to be a `const` type implementing `Allocator`
+since it wouldn't be sound to interact with a static. This may cause confusion
+since the allocator itself will therefor not be allowed to maintain any state
+internally since a new instance will be created for each use. In addition, the
+`Allocator` trait uses a `Layout` type as a higher level encapsulation of the
+requested alignment and size of the allocation. The larger API surface area
+will most likely cause this feature to have a significantly longer stabilization
```

Comment:
I'm not so sure about this any more. At least the piece of the API surface named here (`Layout`) doesn't seem *that* significant, I don't recall any unresolved questions about it (there's questions about what alignment means for some functions, but that's independent of whether it's wrapped in a `Layout` type).